core/trino-main/src/main/java/io/trino/sql/planner/planprinter/IoPlanPrinter.java

```diff
@@ -248,28 +248,85 @@ public boolean equals(Object obj)
                 }
                 TableColumnInfo o = (TableColumnInfo) obj;
                 return Objects.equals(table, o.table) &&
-                        Objects.equals(columnConstraints, o.columnConstraints) &&
+                        Objects.equals(constraint, o.constraint) &&
                         Objects.equals(estimate, o.estimate);
             }
 
             @Override
             public int hashCode()
             {
-                return Objects.hash(table, columnConstraints, estimate);
+                return Objects.hash(table, constraint, estimate);
             }
 
             @Override
             public String toString()
             {
                 return toStringHelper(this)
                         .add("table", table)
-                        .add("columnConstraints", columnConstraints)
+                        .add("constraint", constraint)
                         .add("estimate", estimate)
                         .toString();
             }
         }
     }
 
+    public static class Constraint
+    {
+        private final boolean isNone;
+
+        private final Set<ColumnConstraint> columnConstraints;
+
+        @JsonCreator
+        public Constraint(
+                @JsonProperty("none") boolean isNone,
+                @JsonProperty("columnConstraints") Set<ColumnConstraint> columnConstraints)
+        {
+            this.isNone = isNone;
+            this.columnConstraints = columnConstraints;
```

Comment:
make a defensive copy to ensure that Constraint is immutable

```suggestion
            this.columnConstraints = ImmutableSet.copyIf(requireNonNull(columnConstraints, "columnConstraints is null"));
```